cocos/2d/CCSprite.cpp

```diff
@@ -425,50 +426,165 @@ void Sprite::setTextureRect(const Rect& rect, bool rotated, const Size& untrimme
+                            _rect.size.width,
+                            _rect.size.height);
+        }
+        setVertexCoords(copyRect, &_quad);
         _polyInfo.setQuad(&_quad);
     } else {
+        // case C)
+
         // in theory it can support 3 slices as well, but let's stick to 9 only
         CCASSERT(_numberOfSlices == 9, "Invalid number of slices");
 
+
+        // How the texture is split
+        //
+        //  u,v: are the texture coordinates
+        //  w,h: are the width and heights
+        //
+        //      w0    w1   w2
+        // v2 +----+------+--+
+        //    |    |      |  |
+        //    |    |      |  |
+        //    | 6  |   7  | 8|  h2
+        //    |    |      |  |
+        // v1 +----+------+--|
+        //    |    |      |  |
+        //    | 3  |   4  | 5|  h1
+        // v0 +----+------+--|
+        //    |    |      |  |
+        //    | 0  |   1  | 2|  h0
+        //    |    |      |  |
+        //    +----+------+--+
+        //    u0   u1    u2
+        //
+        //
+        //  and when the texture is rotated, it will get transformed.
+        //  not only the rects have a different position, but also u,v
+        //  points to the bottom-left and not top-right of the texture
+        //  so some swaping/origin/reordering needs to be done in order
+        //  to support rotated slice-9 correctly
+        //
+        //       w0    w1     w2
+        // v2 +------+----+--------+
+        //    |      |    |        |
+        //    |   0  |  3 |    6   | h2
+        // v1 +------+----+--------+
+        //    |      |    |        |
+        //    |   1  |  4 |    7   | h1
+        //    |      |    |        |
+        // v0 +------+----+--------+
+        //    |   2  |  5 |    8   | h0
+        //    +------+----+--------+
+        //    u0      u1     u2
+
+
         // center rect
-        const float cx1 = _centerRectNormalized.origin.x;
-        const float cy1 = _centerRectNormalized.origin.y;
-        const float cx2 = _centerRectNormalized.origin.x + _centerRectNormalized.size.width;
-        const float cy2 = _centerRectNormalized.origin.y + _centerRectNormalized.size.height;
+        float cx1 = _centerRectNormalized.origin.x;
+        float cy1 = _centerRectNormalized.origin.y;
+        float cx2 = _centerRectNormalized.origin.x + _centerRectNormalized.size.width;
+        float cy2 = _centerRectNormalized.origin.y + _centerRectNormalized.size.height;
 
         // "O"riginal rect
         const float oox = _rect.origin.x;
         const float ooy = _rect.origin.y;
         const float osw = _rect.size.width;
         const float osh = _rect.size.height;
 
+        if (_rectRotated) {
+            std::swap(cx1, cy1);
+            std::swap(cx2, cy2);
+
+            // when the texture is rotated, then the centerRect starts from the "bottom" (left)
+            // but when it is not rotated, it starts from the top, so invert it
```

Comment:
> but when it is not rotated, it starts from the top

It is only true for get spritframe from plist? Because i think in normal, the it should be start from bottom left.